### ISSUE TITLE
Fine-tune the LLM for mnemonics

### ISSUE THREAD

**Issue by StephanAkkerman** [StephanAkkerman/FluentAI#123]
1. Description:

   - Problem:
     Currently we use a very general LLM for generating the mnemonics. However, we are using 1 template for input sentences and want 1 sentence as output. Maybe we can make it more efficient by finetuning the LLM.

   - Solution:
     Look into the process of finetuning LLMs and see if it's possible for our use case. 

   - Prerequisites:
     [List any requirements or dependencies needed before starting.]

2. Tasks:
   - Research finetuning of LLM
   - Find out if this works for us
   - Train the model (or add a new issue for this)

3. Additional context

**Comment by StephanAkkerman** [comment]
https://www.reddit.com/r/LocalLLaMA/comments/1fm59kg/how_do_you_actually_finetune_a_llm_on_your_own/?rdt=61769

**Comment by TimKoornstra** [comment]
Or use LORA

**Comment by StephanAkkerman** [comment]
https://www.reddit.com/r/MachineLearning/s/Vj23oXeGuu

**Comment by StephanAkkerman** [comment]
Like at https://huggingface.co/spaces/k-mktr/gpu-poor-llm-arena for some more models

**Comment by StephanAkkerman** [comment]
Simple guide on how to implement LORAs with huggingface: https://medium.com/@kram254/train-large-language-models-with-lora-and-huggingface-d04be693dd7a

**Comment by StephanAkkerman** [comment]
Could make our own dataset scraping sources like:
- https://forum.artofmemory.com/t/mnemonics-for-spanish-vocabulary/34816 from https://artofmemory.com/language-learning/
- https://mnemonicdictionary.com/wordlist/GREwordlist

**Comment by StephanAkkerman** [comment]
Plan for generating the dataset:
- Scrape sources for mnemonics
- Ask SOTA model to create the sentence
- Save this dataset

**Comment by StephanAkkerman** [comment]
Maybe use linkword for training: https://en.wikipedia.org/wiki/Linkword

**Comment by StephanAkkerman** [comment]
Use a LLM that has good results for creative writing:
- https://eqbench.com/creative_writing.html
- https://www.nc-bench.com/models

**Comment by StephanAkkerman** [comment]
Done and will be integrated with #210
